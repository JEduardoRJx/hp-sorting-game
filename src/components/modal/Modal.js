import React from 'react';
import './Modal.scss'

const Modal = ({closeModal}) => {
  return (
    <div className='modal'>
      <div className='modal-contents'>
        <h1 className='modal-title'>Harry Potter and the Sorting Game</h1>
        <h2 className='click-me'>Click Below to Begin</h2>
        <img className ='modal-img' 
          src={require('../../images/marauders_map.png')} 
          alt='marauders map logo' 
          onClick={closeModal}/>
      </div>
      <img className ='modal-gif' 
        src={require('../../images/issiautng.gif')} 
        alt='I solemnly swear I am up to no good gif'/>
    </div>
  )
}

export default Modal;