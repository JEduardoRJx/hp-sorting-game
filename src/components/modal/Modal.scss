$hw-dark: #4c3638;
$hw-light: #D3A481;

.modal {
  background-color: $hw-light;
  color: $hw-dark;
  height: 100%;
  width: 100%;
  z-index: 1;
  position: fixed;
  top: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.modal-contents {
  width: 100%;
  text-align: center;
}

.modal-title {
  font-family: 'harry_pregular';
  font-size: 6em;
  color: $hw-dark;
}

.modal-img {
  height: 300px;
  margin-top: 30px;
  &:hover {
    transform: scale(1.2);
    cursor: pointer;
  }
}

.modal-gif {
  display: none;
}