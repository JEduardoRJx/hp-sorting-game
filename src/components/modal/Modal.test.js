import React from 'react';
import { shallow } from 'enzyme';
import Modal from './Modal'

describe('MainScreen', () => {
  let wrapper;
  const mockCloseModal = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<Modal
      closeModal={mockCloseModal} />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call mockCloseModal when image is clicked', () => {
    wrapper.find('.modal-img').simulate('click');
    expect(mockCloseModal).toHaveBeenCalled()
  });
})