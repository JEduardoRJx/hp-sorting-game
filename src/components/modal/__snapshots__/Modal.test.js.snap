// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen should match snapshot 1`] = `
<div
  className="modal"
>
  <div
    className="modal-contents"
  >
    <h1
      className="modal-title"
    >
      Harry Potter and the Sorting Game
    </h1>
    <h2
      className="click-me"
    >
      Click Below to Begin
    </h2>
    <img
      alt="marauders map logo"
      className="modal-img"
      onClick={[MockFunction]}
      src="marauders_map.png"
    />
  </div>
  <img
    alt="I solemnly swear I am up to no good gif"
    className="modal-gif"
    src="issiautng.gif"
  />
</div>
`;
